### PR TITLE
Add new Altoida schemas

### PR DESCRIPTION
**Pull request by mpgxvii**
- Add Altoida xml (new `altoida_data.xml` file) metadata schemas

### CODE REVIEW

File: commons/connector/upload/altoida/altoida_contrast_vision_test.avsc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "type": "array",
+        "items": {
+          "name": "ContrastCircle",
+          "type": "record",
+          "doc": "Contrast circle screen element.",
+           "fields": [
+             { "name": "id", "type": "string", "doc": "Circle id.", "default": null },
+             { "name": "radius", "type": "double", "doc": "Circle radius" },
+             { "name": "xCenterOffset", "type": "double", "doc": "Circle location x center offset." },
+             { "name": "yCenterOffset", "type": "double", "doc": "Circle location y center offset." },
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_contrast_vision_test.avsc:20]: what units is this in?

File: commons/connector/upload/altoida/altoida_contrast_vision_test.avsc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+             { "name": "color", "type": ["null", "string"], "doc": "Circle color." }
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_contrast_vision_test.avsc:21]: what value type does this have? text? hex rgb?

File: commons/connector/upload/altoida/altoida_contrast_vision_test.avsc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+           ]
+         }}, "doc": "Assessment events list."},
+      { "name": "events", "type": {
+        "type": "array",
+        "items": {
+          "name": "AssessmentEvent",
+          "type": "record",
+          "doc": "Assessment event.",
+           "fields": [
+             { "name": "eventType", "type": ["null", "string"], "doc": "Event type.", "default": null },
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_contrast_vision_test.avsc:31]: what are some possible values?

File: commons/connector/upload/altoida/altoida_contrast_vision_test.avsc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+             { "name": "timestamp", "type": "double", "doc": "Timestamp in UTC (s) when the event occurred." },
+             { "name": "location", "type": ["null", "string"], "doc": "Tapping location." }
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_contrast_vision_test.avsc:33]: what are some possible values?

File: commons/connector/upload/altoida/altoida_motor_bubble.avsc [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "namespace": "org.radarcns.connector.upload.altoida",
+  "type": "record",
+  "name": "AltoidaMotorBubble",
+  "doc": "Data from Altoida motor bubble test.",
+  "fields": [
+    {"name": "time", "type": "double", "doc": "Timestamp in unix time acquired locally from the device."},
+    {"name": "timeReceived", "type": "double", "doc": "Timestamp in unix time received from Altoida API."},
+    {"name": "x", "type": "double", "doc": "Motor bubble x."},
+    {"name": "y", "type": "double", "doc": "Motor bubble y."}
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_motor_bubble.avsc:10]: What are the units?

File: commons/connector/upload/altoida/altoida_tapping_test.avsc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      { "name": "events", "type": {
+        "type": "array",
+        "items": {
+          "name": "AssessmentEvent",
+          "type": "record",
+          "doc": "Assessment event.",
+           "fields": [
+             { "name": "eventType", "type": ["null", "string"], "doc": "Event type.", "default": null },
+             { "name": "timestamp", "type": "double", "doc": "Timestamp in UTC (s) when the event occurred." },
+             { "name": "location", "type": ["null", "string"], "doc": "Tapping location." }
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_tapping_test.avsc:19]: what are some possible values?

File: commons/connector/upload/altoida/altoida_xml_metadata.avsc [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+      "type": "double",
+      "doc": "Timestamp in unix time acquired locally from the device."
+    },
+    {
+      "name": "timeReceived",
+      "type": "double",
+      "doc": "Timestamp in unix time received from Altoida API."
+    },
+    {
+      "name": "age",
```
[the remark this answers is not in the document]
Comment (blootsvoets) [reply commons/connector/upload/altoida/altoida_xml_metadata.avsc:18]: Just one schema should be fine.

**Review by blootsvoets — COMMENTED** [review]
LGTM, though it would be good to get clarification on some of the units and ranges.

File: commons/connector/upload/altoida/altoida_contrast_vision_test.avsc (hunk 2 above)
Comment (mpgxvii) [reply commons/connector/upload/altoida/altoida_contrast_vision_test.avsc]: It's RGB. I've made the changes.

File: commons/connector/upload/altoida/altoida_ar_test.avsc [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+             { "name": "xCenterOffset", "type": "double", "doc": "Element location x center offset in cm." },
+             { "name": "yCenterOffset", "type": "double", "doc": "Element location y center offset in cm." },
+             { "name": "width", "type": "double", "doc": "Element width in cm." },
+             { "name": "height", "type": "double", "doc": "Element height in cm." }
+           ]
+         }}, "doc": "Buttons / screen elements list."},
+      { "name": "events", "type": {
+        "type": "array",
+        "items": {
+          "name": "AssessmentEvent",
```
Comment (mpgxvii) [inline commons/connector/upload/altoida/altoida_ar_test.avsc:27]: @blootsvoets There are a few schemas that have the `events` (`AssessmentEvent`) array data. Should they just share the same data type or is it fine that they're separate?

Comment (blootsvoets) [reply commons/connector/upload/altoida/altoida_ar_test.avsc:27]: If they have the same elements (or very similar), it's fine to make it a shared object. Preferably in a separate file as well then.

**Review by blootsvoets — COMMENTED** [review]
When mapping arrays to CSV, each array item will become a column. On CSV column mismatches, new files will be created. If the number of events is not fixed, the events could rather be sent to a separate topic and associated with a shared ID. Similarly for other array structures (circles, buttons).

**Comment by mpgxvii** [comment]
> When mapping arrays to CSV, each array item will become a column. On CSV column mismatches, new files will be created. If the number of events is not fixed, the events could rather be sent to a separate topic and associated with a shared ID. Similarly for other array structures (circles, buttons).

Does that mean we should separate each event into different records instead?

**Comment by blootsvoets** [comment]
> Does that mean we should separate each event into different records instead?

Yes, I think that would make the resulting CSV files much easier to parse.

File: commons/connector/upload/altoida/altoida_contrast_vision_test.avsc [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+          "doc": "Contrast circle screen element.",
+           "fields": [
+             { "name": "id", "type": "int", "doc": "Circle id." },
+             { "name": "radius", "type": "double", "doc": "Circle radius" },
+             { "name": "xCenterOffset", "type": "double", "doc": "Circle location x center offset in cm." },
+             { "name": "yCenterOffset", "type": "double", "doc": "Circle location y center offset in cm." },
+             { "name": "colorAlphaValue", "type": ["null", "string"], "doc": "Circle alpha value in RGBA color space.", "default" : null },
+             { "name": "colorRValue", "type": ["null", "string"], "doc": "Circle red value in RGBA color space.", "default" : null },
+             { "name": "colorGValue", "type": ["null", "string"], "doc": "Circle green value in RGBA color space.", "default" : null },
+             { "name": "colorBValue", "type": ["null", "string"], "doc": "Circle blue value in RGBA color space.", "default" : null }
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_contrast_vision_test.avsc:24]: Are these hexadecimals (00-ff) or can they be float (0.0-1.0) or decimal (0-255) instead?

Comment (mpgxvii) [reply commons/connector/upload/altoida/altoida_contrast_vision_test.avsc:24]: Yes I checked and they are float. I'll make the changes.

**Comment by mpgxvii** [comment]
> > Does that mean we should separate each event into different records instead?
> 
> Yes, I think that would make the resulting CSV files much easier to parse.

Should they be linked by assessment or will the timestamps be sufficient?

**Comment by mpgxvii** [comment]
> > Does that mean we should separate each event into different records instead?
> 
> Yes, I think that would make the resulting CSV files much easier to parse.

I've removed the arrays of events in the schemas, but kept the screen elements arrays since those should be fixed for each assessment type. Is that ok?

File: commons/connector/upload/altoida/altoida_assessments_summary.avsc [hunk 10]
```diff
@@ -0,0 +1,15 @@
+{
+    "namespace": "org.radarcns.connector.upload.altoida",
+    "name": "AltoidaAssessmentsSummary",
+    "type": "record",
+    "doc": "Summary timestamps of a completed Altoida assessment.",
+    "fields": [
+      { "name": "time", "type": "double", "doc": "Timestamp in UTC (s) when the assessment is started by the subject." },
+      { "name": "timeReceived", "type": "double", "doc": "Timestamp in unix time received from Altoida API."},
+      { "name": "assessmentName", "type": ["null", "string"], "doc": "Assessment name.", "default": null },
+      { "name": "startTime", "type": "double", "doc": "Timestamp in UTC (s) when the assessment is started." },
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_assessments_summary.avsc:10]: The description is identical to “time”, is there a difference?

File: commons/connector/upload/altoida/altoida_assessment_event.avsc [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{
+  "namespace": "org.radarcns.connector.upload.altoida",
+  "type": "record",
+  "name": "AltoidaAssessmentEvent",
+  "doc": "Altoida assessmment event.",
+  "fields": [
+    { "name": "assessmentName", "type": ["null", "string"], "doc": "Assessment name.", "default": null },
+    { "name": "assessmentTimestamp", "type": "double", "doc": "Timestamp in UTC (s) when the assessment is started." },
+    { "name": "eventType", "type": ["null", "string"], "doc": "Event type.", "default": null },
+    { "name": "timestamp", "type": "double", "doc": "Timestamp in UTC (s) when the event occurred." },
```
Comment (blootsvoets) [inline commons/connector/upload/altoida/altoida_assessment_event.avsc:10]: Rename to `time` and make first field?

File: commons/connector/upload/altoida/altoida_assessments_summary.avsc (hunk 10 above)
Comment (mpgxvii) [reply commons/connector/upload/altoida/altoida_assessments_summary.avsc:10]: Yeah they are identical. I can remove `time`.

**Review by blootsvoets — APPROVED** [review]
LGTM

**Pull request merged**